code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -4,70 +4,15 @@
 		M << "No attacking people at spawn, you jackass."
 		return
 
-	var/datum/organ/external/temp = M:organs_by_name["r_hand"]
-	if (M.hand)
-		temp = M:organs_by_name["l_hand"]
-	if(temp && !temp.is_usable())
-		M << "<span class='warning'>You can't use your [temp.display_name].</span>"
-		return
-
-	..()
+	..() //Checks hand validity and shares contact diseases; also exits if attacker not carbon
 
 	if((M != src) && check_shields(0, M.name))
 		visible_message("<span class='danger'>[M] attempted to touch [src]!</span>")
 		return 0
 
 
-	if(M.gloves && istype(M.gloves,/obj/item/clothing/gloves))
-		var/obj/item/clothing/gloves/G = M.gloves
-		if(G.cell)
-			if(M.a_intent == I_HURT)//Stungloves. Any contact will stun the alien.
-				if(G.cell.charge >= 2500)
-					G.cell.use(2500)
-					visible_message("<span class='danger'>[src] has been touched with the stun gloves by [M]!</span>")
-					M.attack_log += text("\[[time_stamp()]\] <font color='red'>Stungloved [src.name] ([src.ckey])</font>")
-					src.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been stungloved by [M.name] ([M.ckey])</font>")
-					if(!iscarbon(M))
-						LAssailant = null
-					else
-						LAssailant = M
-
-					log_attack("<font color='red'>[M.name] ([M.ckey]) stungloved [src.name] ([src.ckey])</font>")
-
-					var/armorblock = run_armor_check(M.zone_sel.selecting, "energy")
-					apply_effects(5,5,0,0,5,0,0,armorblock)
-					return 1
-				else
-					M << "<span class='warning'>Not enough charge! </span>"
-					visible_message("<span class='danger'>[src] has been touched with the stun gloves by [M]!</span>")
-				return
-
-		if(istype(M.gloves , /obj/item/clothing/gloves/boxing/hologlove))
-
-			var/damage = rand(0, 9)
-			if(!damage)
-				playsound(loc, 'sound/weapons/punchmiss.ogg', 25, 1, -1)
-				visible_message("<span class='danger'>[M] has attempted to punch [src]!</span>")
-				return 0
-			var/datum/organ/external/affecting = get_organ(ran_zone(M.zone_sel.selecting))
-			var/armor_block = run_armor_check(affecting, "melee")
-
-			if(M_HULK in M.mutations)			damage += 5
-
-			playsound(loc, "punch", 25, 1, -1)
-
-			visible_message("<span class='danger'>[M] has punched [src]!</span>")
-
-			apply_damage(damage, HALLOSS, affecting, armor_block)
-			if(damage >= 9)
-				visible_message("<span class='danger'>[M] has weakened [src]!</span>")
-				apply_effect(4, WEAKEN, armor_block)
-
-			return
-	else
-		if(istype(M,/mob/living/carbon))
-//			log_debug("No gloves, [M] is truing to infect [src]")
-			M.spread_disease_to(src, "Contact")
+	if(!M.gloves && istype(M,/mob/living/carbon))
+		M.spread_disease_to(src, "Contact")

```

Comment:
should prolly also add ```src.spread_disease_to(M, "Contact")```, and remove the typecheck here (this proc is only supposed to take ````/mob/living/carbon/human``` as M, if it's getting other shit then add a guard at the start)